src/test/Conjure/ParsePrint.hs

```diff
@@ -30,85 +38,121 @@ tests = do
-        fixBackslashes ('\\'        : xs) = '/'    : fixBackslashes xs
+        fixBackslashes ('/' : '\\' : xs) = "/\\" ++ fixBackslashes xs
+        fixBackslashes ('\\' : '/' : xs) = "\\/" ++ fixBackslashes xs
+        fixBackslashes ('\\' : xs) = '/' : fixBackslashes xs
         fixBackslashes [] = []
-        fixBackslashes (x:xs) = x : fixBackslashes xs
+        fixBackslashes (x : xs) = x : fixBackslashes xs
 
         readIfExists :: FilePath -> IO String
         readIfExists f = fromMaybe "" <$> readFileIfExists f
-
-    step "Checking stderr"
-    stderrG <- fixWindowsPaths <$> readIfExists (tBaseDir </> "stderr")
-    stderrE <- readIfExists (tBaseDir </> "stderr.expected")
-    unless (stderrE == stderrG) $
-        assertFailure $ renderNormal $ vcat [ "unexpected stderr:" <++> pretty stderrG
-                                            , "was expecting:    " <++> pretty stderrE ]
-    step "Checking stdout"
-    stdoutG <- fixWindowsPaths <$> readIfExists (tBaseDir </> "stdout")
-    stdoutE <- readIfExists (tBaseDir </> "stdout.expected")
-    unless (stdoutE == stdoutG) $
-        assertFailure $ renderNormal $ vcat [ "unexpected stdout:" <++> pretty stdoutG
-                                            , "was expecting:    " <++> pretty stdoutE ]
-
+    do
+        step "Checking stderr"
+        stderrG <- fixWindowsPaths <$> readIfExists (tBaseDir </> "stderr")
+        stderrE <- readIfExists (tBaseDir </> "stderr.expected")
+        unless (stderrE == stderrG) $
+            assertFailure $
+                renderNormal $
+                    vcat
+                        [ "unexpected stderr:" <++> pretty stderrG
+                        , "was expecting:    " <++> pretty stderrE
+                        ]
+    do
+        step "Checking Generated Representation"
+        stdoutG <- fixWindowsPaths <$> readIfExists (tBaseDir </> "model.json")
+        stdoutE <- readIfExists (tBaseDir </> "model.expected.json")
+        let diffs = do
+                jGiven <- stringToJson stdoutG
+                jReference <- stringToJson stdoutE
+                let Patch ds = diff jGiven jReference
+                return ds
+        case diffs of
+            Nothing -> assertFailure $ "JSON parser error in" ++ stdoutE
+            Just [] -> return ()
+            Just ops -> assertFailure $ renderNormal $ vcat ["Difference in json:" <++> vcat (map (stringToDoc . show) ops)]
+        
+    do
+        step "Checking stdout"
+        stdoutG <- fixWindowsPaths <$> readIfExists (tBaseDir </> "stdout")
+        stdoutE <- readIfExists (tBaseDir </> "stdout.expected")
+        unless (stdoutE == stdoutG) $
+            assertFailure $
+                renderNormal $
+                    vcat
+                        [ "unexpected stdout:" <++> pretty stdoutG
+                        , "was expecting:    " <++> pretty stdoutE
+                        ]
+    do
+        step "Checking Types"
```

Comment:
I see you added this additional test, good idea! We had the typeCheckAllTests as well, but that was checking essence files in the exhaustive directory only.